enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,562 @@
+here too, but that is not required for the initial implementation.
+
+An example of a use case this is not intended to address would be a
+single-nic deployment on a cloud platform with no unusual network settings.
+In that case manual configuration of br-ex would be an unnecessary extra
+burden on the user and the logic to handle it automatically would be fairly
+simple so using configure-ovs is far less likely to cause problems.
+
+## Proposal
+
+This mechanism will be similar to the day-1 network configuration described
+in enhancements/network/baremetal-ipi-network-configuration.md. However,
+there will be some important differences, which will be discussed below.
+
+At a high level, we will provide an interface for the deployer to specify
+per-host interface configuration. Configuration for an OpenVSwitch bridge
+named br-ex will be mandatory. The format for the configuration will be
+NMState YAML, which will be provided to each node via Ignition. Around the
+same point in the boot process that configure-ovs would run, we will run
+NMState to apply the provided configuration. This will only be done on first
+boot because subsequent changes will be made using the Kubernetes-NMState
+operator.
+
+### Important differences from baremetal day-1
+
+Instead of pre-processing the NMState config into NetworkManager
+nmconnection files, we will simply write the NMState file directly to
+the host disk. As of 4.14 we have NMState available at the host level
+so it will no longer be necessary to run NMState in a container.
+
+Since this is intended to be a replacement for configure-ovs, it must
+work in a cross-platform manner. The baremetal feature is implemented in the
+baremetal-operator, which means non-baremetal platforms cannot use it.
+
+However, this is not a replacement for the baremetal feature (and any other
+platform-specific deploy-time network configuration tools). This is because
+minimal networking is required in order to pull Ignition, and since this
+configuration will be provided through Ignition it will still be necessary
+to have platform-specific methods of injecting network config before Ignition
+runs. However, the platform-specific part can be much simpler because it
+won't need to handle every possible network configuration, just enough to
+reach the Ignition endpoint. In some use cases, platform-specific configuration
+won't even be needed.
+
+### Workflow Description
+
+When the user is populating install-config they will provide the necessary
+configuration in the networking section (TODO: Is this the right place?).
+It will look something like this:
+
+networking:
+  networkType: OVNKubernetes
+  machineNetwork:
+  - cidr: 192.0.2.0/24
+  [...]
+  hostConfig:
+  - name: master-0
+    networkConfig:
+      interfaces:
+      - name: enp2s0
+        type: ethernet
+        state: up
+        ipv4:
+          enabled: false
+        ipv6:
+          enabled: false
+      - name: br-ex
```

Comment:
I don't know if it is relevant for this design, but in case the configure-ovs script today also needs to configure ovn-bridge-mappings you may find this useful - we have recently introduced a nice API for it to nmstate: https://bugzilla.redhat.com/show_bug.cgi?id=2209690

With this, you could configure the mapping easily from this CR:

```yaml
hostConfig:
  ovn:
    bridge-mappings:
      - localnet: FOO
        bridge: br-ex
```

Reply:
Yeah, that's definitely a nice fit for this. It would be a pain to introduce an OpenShift-specific API to do that, but with NMState we get it for "free". :-)